extensions/interactions/TextInput/TextInput.js

```diff
@@ -81,3 +82,56 @@ oppia.directive('oppiaShortResponseTextInput', [
     };
   }
 ]);
+
+oppia.factory('textInputRulesService', ['$filter', function($filter) {
+  return {
+    Equals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      return normalizedAnswer.toLowerCase() == normalizedInput.toLowerCase();
+    },
+    FuzzyEquals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var answerString = normalizedAnswer.toLowerCase();
+
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      var inputString = normalizedInput.toLowerCase();
+
+      if(inputString == answerString) return true;
+      var d = [];
+      for(var i = 0; i <= inputString.length; i++){
+        d.push([i]);
+      }
+      for(var j = 0; j <= answerString.length; j++){
+        d[0].push(j);
+      }
+      for(i = 1; i <= inputString.length; i++){
```

Comment:
Also, I'm slightly concerned about this loop, because it `d` doesn't contain the edit distance. The recurrence also produces wrong results on other inputs. (e.g. 'hello' and 'abcde' returns 5). I think you need to get the maximum over substitution, addition and deletion, instead of short-circuiting at substitution.